driver/mounter.go

```diff
@@ -147,7 +148,31 @@ func (m *mounter) Mount(source, target, fsType string, opts ...string) error {
 		return errors.New("target is not specified for mounting the volume")
 	}
 
-	mountArgs = append(mountArgs, "-t", fsType)
+	// This is a raw block device mount. Create the mount point as a file
+	// since bind mount device node requires it to be a file
+	if fsType == "" {
+		// create directory for target, os.Mkdirall is noop if directory exists
+		err := os.MkdirAll(filepath.Dir(target), 0750)
+		if err != nil {
+			return fmt.Errorf("failed to create target directory for raw block bind mount: %v", err)
+		}
+
+		file, err := os.OpenFile(target, os.O_CREATE, 0666)
+		if file != nil {
+			file.Close()
```

Comment:
If the `err` is non-nil, then `file` will always be `nil`. 

```
https://golang.org/src/os/file.go
func OpenFile(name string, flag int, perm FileMode) (*File, error) {
	testlog.Open(name)
	f, err := openFileNolog(name, flag, perm)
	if err != nil {
		return nil, err
	}
	f.appendMode = flag&O_APPEND != 0

	return f, nil
}
```

I opted to check `file != nil` because it read more explicitly since we always want the file closed if we found it since we are only concerned of its existence. It could also be written as follows if you think it is better in terms of readibility:

```
...
if err != nil {
	return fmt.Errorf("failed to create target file for raw block bind mount: %v", err)
}
file.Close()
...
```

Reply:
I personally prefer the style to check the error first because it seems more Go idiomatic and saves us one `if`. Practically, it shouldn't make any difference, so I'm happy to let you make the call.